### PR TITLE
pkg/k8s: add support for IPBlock and Egress Rules with IPBlock

### PR DESCRIPTION
**Pull request by ianvernon**
Signed-off by: Ian Vernon <ian@cilium.io>

Fixes: #1487

### CODE REVIEW

File: pkg/k8s/network_policy.go [hunk 1]
```diff
@@ -104,6 +109,22 @@ func ParseNetworkPolicy(np *networkingv1.NetworkPolicy) (api.Rules, error) {
 		ingresses = append(ingresses, ingress)
 	}
 
+	for _, eRule := range np.Spec.Egress {
+		egress := api.EgressRule{}
+		if eRule.To != nil && len(eRule.To) > 0 {
+			for _, rule := range eRule.To {
+				if rule.NamespaceSelector != nil || rule.PodSelector != nil {
+					// TODO: GH-2095
+					log.Warningf("cilium does not support PodSelector or NamespaceSelector for K8s Egress rules")
```
Comment (tgraf) [inline pkg/k8s/network_policy.go]: 1. `Warning(`
2. Capitalize Cilium

File: pkg/k8s/network_policy.go [hunk 2]
```diff
@@ -84,6 +86,9 @@ func ParseNetworkPolicy(np *networkingv1.NetworkPolicy) (api.Rules, error) {
 				if err != nil {
 					return nil, err
 				}
+				if rule.IPBlock != nil {
+					ingress.FromCIDRSet = []api.CIDRRule{ipBlockToCIDRRule(rule.IPBlock)}
```
Comment (tgraf) [inline pkg/k8s/network_policy.go]: Is this meant to use `append()`?

File: pkg/k8s/network_policy.go [hunk 3]
```diff
@@ -104,6 +109,22 @@ func ParseNetworkPolicy(np *networkingv1.NetworkPolicy) (api.Rules, error) {
...
+	for _, eRule := range np.Spec.Egress {
+		egress := api.EgressRule{}
+		if eRule.To != nil && len(eRule.To) > 0 {
+			for _, rule := range eRule.To {
+				if rule.NamespaceSelector != nil || rule.PodSelector != nil {
+					// TODO: GH-2095
+					log.Warningf("cilium does not support PodSelector or NamespaceSelector for K8s Egress rules")
+				}
+				if rule.IPBlock != nil {
+					egress.ToCIDRSet = []api.CIDRRule{ipBlockToCIDRRule(rule.IPBlock)}
```
Comment (tgraf) [inline pkg/k8s/network_policy.go]: Is this meant to use `append()`?

File: pkg/k8s/network_policy_test.go [hunk 4]
```diff
@@ -956,3 +956,98 @@ func (s *K8sSuite) TestNetworkPolicyExamples(c *C) {
...
+    "podSelector": {
+      "matchLabels": {
+        "role": "backend"
+      }
+    },
+    "ingress": [
+      {
+        "from": [
+          {
+			"namespaceSelector": {
```
Comment (tgraf) [inline pkg/k8s/network_policy_test.go]: Indentation

File: pkg/k8s/network_policy_test.go [hunk 5]
```diff
@@ -956,3 +956,98 @@ func (s *K8sSuite) TestNetworkPolicyExamples(c *C) {
...
+	c.Assert(rules, NotNil)
+	c.Assert(len(rules), Equals, 1)
+	c.Assert(rules[0].Egress[0].ToCIDRSet[0].Cidr, Equals, api.CIDR("10.0.0.0/8"))
+
+	expectedCIDRs := []api.CIDR{"10.96.0.0/12", "10.255.255.254/32"}
+
+	for k, v := range rules[0].Egress[0].ToCIDRSet[0].ExceptCIDRs {
+		c.Assert(v, Equals, expectedCIDRs[k])
+	}
+	//c.Assert(rules[0].Egress[0].ToCIDRSet[0].ExceptCIDRs, Equals, []api.CIDR{"10.96.0.0/12", "10.255.255.254/32"})
```
Comment (tgraf) [inline pkg/k8s/network_policy_test.go]: Remove this

**Review by tgraf — DISMISSED** [review]

File: pkg/k8s/network_policy.go (hunk 2 above)
Comment (ianvernon) [reply pkg/k8s/network_policy.go]: You are correct; fixed.

File: pkg/k8s/network_policy.go (hunk 1 above)
Comment (ianvernon) [reply pkg/k8s/network_policy.go]: Fixed.

File: pkg/k8s/network_policy_test.go (hunk 4 above)
Comment (ianvernon) [reply pkg/k8s/network_policy_test.go]: Fixed.

File: pkg/k8s/network_policy_test.go (hunk 5 above)
Comment (ianvernon) [reply pkg/k8s/network_policy_test.go]: Fixed.

File: pkg/k8s/network_policy.go (hunk 3 above)
Comment (ianvernon) [reply pkg/k8s/network_policy.go]: You are correct; fixed.

**Review by tgraf — APPROVED** [review]

**Pull request merged**